mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -44,9 +44,11 @@ private[util] sealed trait BaseReadWrite {
 
   /**
    * Sets the Spark SQLContext to use for saving/loading.
+   *
+   * @deprecated Use session instead. This method will be removed in 3.0.0.
```

Comment:
Add this can make Scala ```MLWriter``` and ```MLReader``` produce correct deprecated annotation, just like the screenshot in PR description.